erc20-bridge-token/test/BridgeToken.js

```diff
@@ -298,4 +322,118 @@ describe('BridgeToken', () => {
       .be
       .revertedWith(`AccessControl: account ${user.address.toLowerCase()} is missing role ${ADMIN_ROLE}`);
   })
+
+  describe("Whitelist", function() {
+    let tokenInfo;
+    let testProofId = 1;
+    const recipient = "testrecipient.near";
+
+    beforeEach(async function() {
+      tokenInfo = await createEmptyToken(
+        nearTokenId,
+        BridgeTokenFactory,
+        BridgeTokenInstance
+      );
+      setMetadata(nearTokenId, tokenInfo.tokenProxyAddress);
+
+      const lockResultProof = require('./proof_template.json');
+      lockResultProof.outcome_proof.outcome.status.SuccessValue = serialize(
+        SCHEMA,
+        "LockResult",
+        {
+          prefix: RESULT_PREFIX_LOCK,
+          token: nearTokenId,
+          amount: 100,
+          recipient: ethers.utils.arrayify(adminAccount.address),
+        }
+      ).toString("base64");
+      const source = "123456789FGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+      lockResultProof.outcome_proof.outcome.receipt_ids[0] = source[
+        testProofId
+      ].repeat(44);
+      testProofId += 1;
+      await BridgeTokenFactory.deposit(
+        borshifyOutcomeProof(lockResultProof),
+        1090
+      );
+    });
+
+    it("Test account in whitelist", async function() {
+      await BridgeTokenFactory.setTokenWhitelistMode(nearTokenId, 3);
+      await BridgeTokenFactory.addAccountToWhitelist(
+        nearTokenId,
+        adminAccount.address
+      );
+      await BridgeTokenFactory.withdraw(nearTokenId, 100, recipient);
+      expect(
+        (await tokenInfo.token.balanceOf(adminAccount.address)).toString()
+      ).to.be.equal("0");
+    });
+
+    it("Test token in whitelist", async function() {
+      await BridgeTokenFactory.setTokenWhitelistMode(nearTokenId, 2);
+      await BridgeTokenFactory.withdraw(nearTokenId, 100, recipient);
+      expect(
+        (await tokenInfo.token.balanceOf(adminAccount.address)).toString()
+      ).to.be.equal("0");
+    });
```

Comment:
Let's also add a test that will use several tokens, whitelist one of them, and verify that one works as whitelisted and the other does not. Then the same with some accounts on these tokens.

(Please add the same to Rust tests)

Reply:
0db3db48a358b519e9ff357dba4729856a3c0cb3
and 
d988be855d875e4031f22cfc3a75f4638e305503